pkg/api/v1alpha1/controlplaneupgrade_types.go

```diff
@@ -9,14 +10,8 @@ const ControlPlaneUpgradeKind = "ControlPlaneUpgrade"
 
 // ControlPlaneUpgradeSpec defines the desired state of ControlPlaneUpgrade.
 type ControlPlaneUpgradeSpec struct {
-	Cluster                Ref     `json:"cluster"`
-	ControlPlane           Ref     `json:"controlPlane"`
-	MachinesRequireUpgrade []Ref   `json:"machinesRequireUpgrade"`
-	KubernetesVersion      string  `json:"kubernetesVersion"`
-	KubeletVersion         string  `json:"kubeletVersion"`
-	EtcdVersion            *string `json:"etcdVersion,omitempty"`
-	CoreDNSVersion         *string `json:"coreDNSVersion,omitempty"`
-	KubeadmClusterConfig   string  `json:"kubeadmClusterConfig"`
+	ControlPlane corev1.ObjectReference `json:"controlPlane"`
```

Comment:
We may need to keep an object ref to the cluster object too In order to pull the bundle from there?